### PR TITLE
Websocket implementation

### PR DESCRIPTION
**Pull request by pavlosymonov**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
```
Comment (penkova) [inline main.go:12]: Do you understand why we use 1024?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/websocket"
```
Comment (penkova) [inline main.go:7]: please provide me tech details to close the knowledge point: 
1. What is WebSockets?
2. How WebSockets differ from traditional HTTP requests?
3. What is Short Polling? And what is Long Polling?
4. What communication techniques do you know?
5. We can use several libraries for building a WebSocket server. What do you know besides gorilla/websocket library? Why did you choose gorilla/websocket?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+var upgrader = websocket.Upgrader{
```
Comment (penkova) [inline main.go:10]: 6. Why do we need to upgrade the connection?

File: main.go (hunk 1 above)
Comment (pavlosymonov) [reply main.go:12]: 1024 is an even binary number. Computers live binary and their instruction sets are packed with bits, nibbles, bytes, words, and integers. They all fit evenly into 1024 memory locations. Also, It used to save memory, but nowadays this is no longer so important.

File: main.go (hunk 2 above)
Comment (pavlosymonov) [reply main.go:7]: 1. Protocol based on TCP, designed to exchange browser messages with a web server;
2. WebSockets keep a single, persistent connection. HTTP request/response-based protocol.
3. Short Polling - The client continuously makes calls to the resource at a specified interval and the resource responds every time. This repeats in very short cycles. Long Polling is where the client initiates the request to the resource and if the resource doesn't have data it keeps that connection open to respond later but if it does have data it responds right away.
4. Verbal, Non-Verbal, Written, Listening, and Visual communications.

**Review by penkova — APPROVED** [review]

**Pull request merged**